internals/secrethub/service_aws_init.go

```diff
@@ -85,13 +86,19 @@ func (cmd *ServiceAWSInitCommand) Run() error {
 
 	fmt.Fprintf(cmd.io.Stdout(), "Detected access to AWS account %s.", accountID)
 
-	region := sess.Config.Region
-	if isSet(region) {
-		fmt.Fprintf(cmd.io.Stdout(), "Using region %s.", *region)
+	if !isSet(cfg.Region) && cmd.kmsKeyID != "" {
+		kmsARN, err := arn.Parse(cmd.kmsKeyID)
+		if err == nil {
+			cfg = cfg.WithRegion(kmsARN.Region)
+		}
+	}
+
+	if isSet(cfg.Region) {
+		fmt.Fprintf(cmd.io.Stdout(), "Using region %s.", *cfg.Region)
 	}
 	fmt.Fprintln(cmd.io.Stdout())
```

Comment:
This logic is a bit hard to understand. If it is set, you print something, then you print a newline always and else you ask for input (printing again). Is there a way to make this easier to read?